File: python/src/cm_email_webhook_verification/signer.py
import base64
import hashlib
import hmac


def generate_signature(secret: str, payload: str) -> str:
    key_bytes = secret.encode("utf-8")
    payload_bytes = payload.encode("utf-8")

    signature = hmac.new(key_bytes, payload_bytes, hashlib.sha512)
    return base64.b64encode(signature.digest()).decode("utf-8")

File: go/signer.go
package webhook

import (
	"crypto/hmac"
	"crypto/sha512"
	"encoding/base64"
)

func GenerateSignature(secret, payload string) string {
	h := hmac.New(sha512.New, []byte(secret))
	h.Write([]byte(payload))
	return base64.StdEncoding.EncodeToString(h.Sum(nil))
}
